### DISCUSSION TITLE
v0.8.0

### DISCUSSION THREAD

**Discussion by valadaptive** [ntsc-rs/ntsc-rs#139] [General]
# What's new

## macOS support

ntsc-rs finally supports macOS! Thank you to all the macOS users who [helped test things out](https://github.com/valadaptive/ntsc-rs/issues/36). This was my first time developing anything for macOS, so it took quite a while to learn the quirks of the platform--I've been working on this for the past couple of months, and I appreciate your patience.

> [!NOTE]
> The code is currently not signed, which means Gatekeeper really doesn't want to run the installer(s). To install ntsc-rs on macOS, [follow the instructions on the website](https://ntsc.rs/docs/macos-installation/).

![ntsc-rs-macos](https://github.com/user-attachments/assets/41c2b58e-2f79-45e4-83d0-b7c3fd19cc13)

## Preset manager

You can now save and load settings presets directly in the app itself via the new preset manager panel! This should streamline the process of switching between different presets.

https://github.com/user-attachments/assets/762bc72b-eecf-4fc2-9d31-a2d623eaac1c

Currently, the application doesn't come with any presets. I'll continue to look into the best way to bundle them--I don't want users to end up accidentally deleting and overwriting the default presets and not knowing how to get them back, but I also don't want to annoy them with hardcoded presets that they don't find useful and can't hide or remove.

## New default settings

The default settings haven't been updated since the very first version of ntsc-rs. I've updated them to something that should look more accurate out of the box.

## Rewritten After Effects plugin

The After Effects/Premiere plugin has been completely rewritten! The original version was written in C and was difficult to maintain. It saw [a](https://github.com/valadaptive/ntsc-rs/commit/b59dd24c02eadd4afe4e00c0272178f75fea9170) [huge](https://github.com/valadaptive/ntsc-rs/commit/413c25b215fbab48dd2b1b43c15d5588e86b6292) [number](https://github.com/valadaptive/ntsc-rs/commit/0091b3b7912d9f6149cf57de0b3211d316d01c0c) [of](https://github.com/valadaptive/ntsc-rs/commit/a2f9aa2ed9033fa6529359834713f7de9748a67b) [bugs](https://github.com/valadaptive/ntsc-rs/commit/b77f08e2653db57cb67c60ac9ab3bb5894ca4f5c) [and](https://github.com/valadaptive/ntsc-rs/commit/bf9c6bfcca58b593f57cb312d083ff97b59b0044) [regressions](https://github.com/valadaptive/ntsc-rs/commit/66cd8a79a4123d2d825e959ffa8fc076ff2d3506).

The new version of the After Effects plugin is written completely in Rust, thanks to [Rust bindings to the After Effects API](https://github.com/AdrianEddy/after-effects) that weren't available when I first wrote the plugin. It should now be much easier not just to maintain, but to build as well! Building C projects across multiple platforms is already challenging, and the After Effects SDK requires some bespoke build tools. The Rust bindings handle that part as well!

As a user of the plugin, you shouldn't notice any differences in functionality (although I did fix a crash since the last release). If you run into any new bugs, feel free to open an issue, preferably including instructions for reproducing the bug or even an After Effects project file that demonstrates it.

## Other

- A loading spinner now appears while the video preview is loading
- You can now adjust the application's zoom level in the "View" menu
- The volume slider now actually mutes the audio when set at 0%
- The filename of the loaded video is now displayed in the window's title
- Fixed a bug where the "snow" effect was less dense near the left edge of the video
- Some code in the OpenFX plugin has been cleaned up as well. Open an issue if you run into any new bugs

<hr /><em>This discussion was created from the release <a href='https://github.com/valadaptive/ntsc-rs/releases/tag/v0.8.0'>v0.8.0</a>.</em>

**Comment by Degamisu** [comment]
Well done!

**Comment by SplaatSites** [comment]
I can't open ntsc-rs-standalone.exe on my Laptop! What do i do?

**Reply by Degamisu** [reply]
Whats going on with it? is it hanging on load? if so, keep it on for 10 - 15 minutes.
